### PR TITLE
feat(merchant_account_v2): add merchant_account_v2 domain and diesel models

### PR DESCRIPTION
**Pull request by Narayanbhat166**
## Type of Change

- [x] New feature

## Description

This PR adds support for merchant account v2 domain and diesel models. This PR introduces a temporary feature flag `merchant_account_v2` and `payment_v2`. These temporary features help us to ship the v2 features without affecting the overall application. Once all the dependant modules migrate to v2, these feature flags will be removed.

### Additional Changes

- [x] This PR modifies the database schema

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

File: crates/router/src/core/payment_methods/cards.rs [hunk 1]
```diff
@@ -3167,7 +3167,9 @@ pub async fn list_payment_methods(
         .and_then(|bp| bp.collect_billing_details_from_wallet_connector);
     Ok(services::ApplicationResponse::Json(
         api::PaymentMethodListResponse {
-            redirect_url: merchant_account.return_url,
+            redirect_url: business_profile
```
Comment (jarnura) [inline crates/router/src/core/payment_methods/cards.rs:3170]: Confirm with respective team before doing this changes.

Comment (Narayanbhat166) [reply crates/router/src/core/payment_methods/cards.rs:3197]: We are not updating the `return_url` merchant account directly, so the updated field will be in `business_profile`. Also the SDK is not using this field

**Review by jarnura — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]
Dashboard specific changes looks fine.

File: crates/hyperswitch_domain_models/src/merchant_account.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@ use common_utils::{
     errors::{CustomResult, ValidationError},
     ext_traits::ValueExt,
     pii,
-    types::keymanager,
+    types::keymanager::{self},
```
Comment (SanchithHegde) [inline crates/hyperswitch_domain_models/src/merchant_account.rs:8]: Nit:

```suggestion
    types::keymanager,
```

Optional change.

File: crates/hyperswitch_interfaces/Cargo.toml [hunk 3]
```diff
@@ -27,10 +27,10 @@ thiserror = "1.0.58"
...
 
 # First party crates
+masking = { version = "0.1.0", path = "../masking" }
 api_models = { version = "0.1.0", path = "../api_models" }
+hyperswitch_domain_models = { version = "0.1.0", path = "../hyperswitch_domain_models", default-features = false, features = ["v1"] }
 common_enums = { version = "0.1.0", path = "../common_enums" }
 common_utils = { version = "0.1.0", path = "../common_utils" }
-hyperswitch_domain_models = { version = "0.1.0", path = "../hyperswitch_domain_models", default-features = false }
-masking = { version = "0.1.0", path = "../masking" }
 router_derive = { version = "0.1.0", path = "../router_derive" }
```
Comment (SanchithHegde) [inline crates/hyperswitch_interfaces/Cargo.toml:35]: Nit: Please keep these sorted.

File: justfile [hunk 4]
```diff
@@ -8,7 +8,7 @@ fmt_flags := '--all'
 fmt *FLAGS:
     cargo +nightly fmt {{ fmt_flags }} {{ FLAGS }}
 
-check_flags := '--all-features --all-targets'
+check_flags := '--all-targets'
 
 # Check compilation of Rust code
 check *FLAGS:
```
Comment (SanchithHegde) [inline justfile:14]: We'd have to update this recipe as well?

File: justfile [hunk 5]
```diff
@@ -60,11 +107,11 @@ euclid-wasm features='dummy_connector':
...
 
-hack_flags := '--workspace --each-feature --all-targets'
-
-# Check compilation of each cargo feature
-hack:
-    cargo hack check {{ hack_flags }}
+# Check compilation of v2 feature on base dependencies
+v2_intermediate_features := "merchant_account_v2,payment_v2"
+hack_v2:
+    cargo hack check  --feature-powerset --ignore-unknown-features --at-least-one-of "v2 " --include-features "v2" --include-features {{ v2_intermediate_features }} --package "hyperswitch_domain_models" --package "diesel_models" --package "api_models"
```
Comment (SanchithHegde) [inline justfile:113]: Specifying `--feature-powerset` can grow exponentially as and when new features are added/included.

File: scripts/ci-checks.sh [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    '$crates_with_features[] | select(IN("v1"; .features[]) | not ) # Select crates without `v1` feature
+    | "\(.name)" # Print out package name'
+)"
+while IFS= read -r crate; do
+  command="cargo hack check --all-targets --each-feature --package \"${crate}\""
+  all_commands+=("$command")
+done <<< "${crates_without_v1_feature}"
+
+echo "The list of commands that will be executed:"
+printf "%s\n" "${all_commands[@]}"
```
Comment (SanchithHegde) [inline scripts/ci-checks.sh:96]: Let's try to print this out in a separate step on CI, for better debugging. This can be taken up in a separate PR.

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**